sormas-app/app/src/main/java/de/symeda/sormas/app/pathogentest/edit/PathogenTestNewActivity.java

```diff
@@ -0,0 +1,177 @@
+import static de.symeda.sormas.app.core.notification.NotificationType.WARNING;
+
+public class PathogenTestNewActivity extends BaseEditActivity<PathogenTest> {
+
+    public static final String TAG = PathogenTestNewActivity.class.getSimpleName();
+
+    private String sampleUuid = null;
+
+    private AsyncTask saveTask;
+
+    public static void startActivity(Context context, String sampleUuid) {
+        BaseEditActivity.startActivity(context, PathogenTestNewActivity.class, buildBundle(sampleUuid));
+    }
+
+    public static Bundler buildBundle(String caseUuid) {
+        return buildBundle(null, 0).setCaseUuid(caseUuid);
+    }
+
+    @Override
+    public Enum getPageStatus() {
+        return null;
+    }
+
+    @Override
+    protected void onCreateInner(Bundle savedInstanceState) {
+        super.onCreateInner(savedInstanceState);
+        sampleUuid = new Bundler(savedInstanceState).getCaseUuid();
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        new Bundler(outState).setCaseUuid(sampleUuid);
+    }
+
+    @Override
+    protected PathogenTest queryRootEntity(String recordUuid) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    protected PathogenTest buildRootEntity() {
+        // basic instead of reference, because we want to have at least the related person
+        Sample associatedSample = DatabaseHelper.getSampleDao().queryUuid(sampleUuid);
+        return DatabaseHelper.getSampleTestDao().build(associatedSample);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        boolean result = super.onCreateOptionsMenu(menu);
+        getSaveMenu().setTitle(R.string.action_save_case);
+        return result;
+    }
+
+    @Override
+    protected BaseEditFragment buildEditFragment(PageMenuItem menuItem, PathogenTest activityRootData) {
+        BaseEditFragment fragment = PathogenTestNewFragment.newInstance(activityRootData);
+        fragment.setLiveValidationDisabled(true);
+        return fragment;
+    }
+
+    @Override
+    protected int getActivityTitle() {
+        return R.string.heading_pathogen_test_new;
+    }
+
+
+    @Override
+    public void replaceFragment(BaseEditFragment f, boolean allowBackNavigation) {
+        super.replaceFragment(f, allowBackNavigation);
+    }
+
+    @Override
+    public void saveData() {
+
+        if (saveTask != null) {
+            NotificationHelper.showNotification(this, WARNING, getString(R.string.message_already_saving));
+            return; // don't save multiple times
+        }
+
+        final PathogenTest PathogenTestToSave = getStoredRootEntity();
+        PathogenTestNewFragment fragment = (PathogenTestNewFragment) getActiveFragment();
+
+        if (PathogenTestToSave.getLabUser() == null) {
+            PathogenTestToSave.setLabUser(ConfigProvider.getUser());
```

Comment:
This is not a good idea because lab users don't use the mobile app, so the result will always be wrong data; instead, this should result in an error and not allow the data to be saved